packages/plugin-flex/src/commands/flex/plugins/archive/plugin-version.ts

```diff
@@ -24,6 +24,12 @@ export default class FlexPluginsArchivePluginVersion extends ArchiveResource<Plu
     }),
   };
 
+  protected _parsedFlags?: OutputFlags<typeof FlexPluginsArchivePluginVersion.flags>;
+
+  async init(): Promise<void> {
```

Comment:
What's the point of this `init`? Are you trying to do this once because it is async and not do it every time in `get _flags`?

Where is this actually called in the CLI when invoked? If this is something that every command needs to have, then define it as `abstract async init()` in the `FlexPlugin` and invoke it automatically as part of the `run` command.

Reply:
Init is a lifecycle method that is called before run and since run requires flags & args to be parsed before it is invoked, I had to go with this approach. Initially I kept the parsing logic in FlexPlugin run fn but that invoked parseCommand with its own flags object so command failed. Similar issues will pop up if I place init only in FlexPlugin and remove the implementation in individual command classes. I believe we need to implement the init method in every command class for this to work

Reply:
Is `init` automatically called by OClif or it's something we need to do?

Reply:
Yup it is called automatically by oclif. Its a lifecycle method like `run` and it is called after constructor and before run

Reply:
So if this it automatically managed, then we don't need the `get _flags` anymore - that was added because you could have called the `this._flags` yourself at any point and wanted to make sure the parsing is done first. But given `oclif` calls this _automatically_ before the `run` command, then the only place you can call it before your code run is in constructor. Therefore, I think it might be safe to fully remove this. I..e `_parseFlags` just becomes `_flags` and this getter is gone.

Reply:
Fixed in [aad90ec](https://github.com/twilio/flex-plugin-builder/pull/722/commits/aad90ec8b37597656bbfc0141045617a67ac945a)